packages/dd-trace/src/plugins/util/web.js

```diff
@@ -253,8 +264,81 @@ const web = {
   },
 
   // Extract the parent span from the headers and start a new span as its child
-  startChildSpan (tracer, name, headers) {
-    const childOf = tracer.extract(FORMAT_HTTP_HEADERS, headers)
+  startChildSpan (tracer, name, req) {
```

Comment:
I'd move all the changes to a helper function exported from an api gateway specific file then call the function in here with minimal changes.

Reply:
Done.